pkg/rancher-desktop/main/commandServer/settingsValidator.ts

```diff
@@ -492,6 +492,79 @@ export default class SettingsValidator {
     return false;
   }
 
+  protected checkExtensions(
+    mergedSettings: Settings,
+    currentValue: Record<string, string>,
+    desiredValue: any,
+    errors: string[],
+    fqname: string,
+  ): boolean {
+    if (typeof desiredValue !== 'object' || !desiredValue) {
+      errors.push(`${ fqname }: "${ desiredValue }" is not a valid mapping`);
+
+      return false;
+    }
+
+    // makeRE is a tagged template for making regular expressions with /x (i.e.
+    // ignoring any whitespace within the regular expression itself).
+    function makeRE(strings: {raw: readonly string[]}, ...subsitutions: any[]) {
+      const subsitutionSources = subsitutions.map(s => s instanceof RegExp ? s.source : s);
+      const raw = String.raw(strings, ...subsitutionSources);
+      const lines = raw.split(/\r?\n/);
+      // Drop comments at end of line
+      const uncommentedLines = lines.map(line => line.replace(/\s#.*$/, ''));
```

Comment:
So you'll never backslash-escape a `#` I take it

Reply:
It only matches `#` prefixed by whitespace. So if you actually need it in a RE like that you can always write `[#]` to get around it trivially.